sql/log_event.h

```diff
@@ -3002,28 +3004,30 @@ class Rand_log_event: public Log_event
       cache_type= Log_event::EVENT_NO_CACHE;
   }
 #ifdef HAVE_REPLICATION
-  void pack_info(Protocol* protocol);
+  void pack_info(Protocol* protocol) override;
 #endif /* HAVE_REPLICATION */
 #else
-  bool print(FILE* file, PRINT_EVENT_INFO* print_event_info);
+  bool print(FILE* file, PRINT_EVENT_INFO* print_event_info) override;
 #endif
 
   Rand_log_event(const char* buf,
                  const Format_description_log_event *description_event);
   ~Rand_log_event() = default;
-  Log_event_type get_type_code() { return RAND_EVENT;}
-  int get_data_size() { return 16; /* sizeof(ulonglong) * 2*/ }
+  Log_event_type get_type_code() override { return RAND_EVENT;}
+  int get_data_size() override { return 16; /* sizeof(ulonglong) * 2*/ }
 #ifdef MYSQL_SERVER
-  bool write();
+  bool write() override;
+#ifdef HAVE_REPLICATION
+  bool is_part_of_group() override { return 1; }
 #endif
```

Comment:
It would be better to indent nested preprocessor conditions with one space per nesting level after the initial `#`:
```c++
#ifdef MYSQL_SERVER
  bool write() override;
# ifdef HAVE_REPLICATION
  bool is_part_of_group() override { return true; }
# endif
```